submissions/okkkko/js-post-oop/index.js

```diff
@@ -0,0 +1,76 @@
+class Inhabitant {
+  constructor(species, name, gender, saying, friends) {
+    this.species = species;
+    this.name = name;
+    this.gender = gender;
+    this.saying = saying;
+    this.friends = friends;
+  }
+  toString() {
+    const properties = ["species", "name", "gender", "saying", "friends"].map(
+      (prop) => `${prop}: ${this[prop]}`
+    );
+    return properties.join("; ");
+  }
+}
+class Human extends Inhabitant {
+  constructor(name, gender, saying, friends) {
+    super("human", name, gender, saying, friends);
+    this.legs = 2;
+    this.hands = 2;
+  }
+  toString() {
+    return `${super.toString()}; ${["hands", "legs"]
+      .map((prop) => `${prop}: ${this[prop]}`)
+      .join("; ")};`;
```

Comment:
Nested literals are hard to read. Formatting also can add to a better readability.

```suggestion
    return [
        super.toString(),
        ...["hands", "legs"].map(prop => `${prop}: ${this[prop]}`)
      ].join("; ");
```

Reply:
Understood, thank you!

Reply:
If I add legs to the Inhabitant class, there will be only hands in the array. Should I leave it this way or get rid of the map method?

Reply:
Both approaches are ok. I would stick to the pattern with map (due to conversion into a standard `"propName: propValue"`) as less error prone.